crates/sui-bridge/src/types.rs

```diff
@@ -116,18 +116,19 @@ impl BridgeCommittee {
 }
 
 impl CommitteeTrait<BridgeAuthorityPublicKeyBytes> for BridgeCommittee {
-    // Note:
-    // 1. preference is not supported today.
-    // 2. blocklisted members are always excluded.
+    // Note: blocklisted members are always excluded.
     fn shuffle_by_stake_with_rng(
         &self,
-        // preference is not supported today
-        _preferences: Option<&BTreeSet<BridgeAuthorityPublicKeyBytes>>,
+        // `preferences` is used as a flag here to influence the order of validators to be requested
+        //  if `Some(_)`, then we will request validators in the order of the voting power
+        //  if `None`, we still refer to voting power, but they are shuffled by randomness.
+        //  to save gas cost.
+        preferences: Option<&BTreeSet<BridgeAuthorityPublicKeyBytes>>,
```

Comment:
is this truly used as a flag? why the `BTreeSet`? are we going to use it later? I am a bit lost here :)

Reply:
this is a trait interface. Here we are indeed using it as a boolean to indicate if we want to order it in one way or another (starting from line 150)